extensions/pkg/controller/backupentry/mapper_test.go

```diff
@@ -0,0 +1,196 @@
+
+		backupEntry  *extensionsv1alpha1.BackupEntry
+		backupEntry2 *extensionsv1alpha1.BackupEntry
+	)
+
+	BeforeEach(func() {
+		fakeClient = fakeclient.NewClientBuilder().WithScheme(kubernetes.SeedScheme).Build()
+		ctrl = gomock.NewController(GinkgoT())
+		cache = mockcache.NewMockCache(ctrl)
+		mgr = mockmanager.NewMockManager(ctrl)
+		mgr.EXPECT().GetCache().Return(cache).AnyTimes()
+
+		namespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-namespace",
+				Annotations: map[string]string{
+					v1beta1constants.ShootUID: "xyz",
+				},
+			},
+		}
+
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: namespace.Name,
+			},
+		}
+
+		backupEntry = &extensionsv1alpha1.BackupEntry{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-namespace--xyz",
+			},
+			Spec: extensionsv1alpha1.BackupEntrySpec{
+				SecretRef: corev1.SecretReference{
+					Name:      secret.Name,
+					Namespace: secret.Namespace,
+				},
+			},
+		}
+		backupEntry2 = &extensionsv1alpha1.BackupEntry{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "backupEntry-2",
+			},
+			Spec: extensionsv1alpha1.BackupEntrySpec{
+				SecretRef: corev1.SecretReference{
+					Name:      secret.Name,
+					Namespace: secret.Namespace,
+				},
+			},
+		}
+	})
+
+	Describe("#SecretToBackupEntryMapper", func() {
+		var mapper mapper.Mapper
+
+		BeforeEach(func() {
+			mapper = SecretToBackupEntryMapper(nil)
+		})
+
+		It("should find all objects for the passed cluster", func() {
+			Expect(fakeClient.Create(ctx, backupEntry)).To(Succeed())
+			Expect(fakeClient.Create(ctx, backupEntry2)).To(Succeed())
+
+			Expect(mapper.Map(ctx, logr.Discard(), fakeClient, secret)).To(ConsistOf(
+				reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name: backupEntry.Name,
+					},
+				},
+				reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name: backupEntry2.Name,
+					},
+				}))
+		})
+
+		It("should find no objects for the passed cluster because predicates do not match", func() {
+			predicates := []predicate.Predicate{
+				predicate.Funcs{
+					GenericFunc: func(event event.GenericEvent) bool {
+						return false
+					},
+				},
+			}
+			mapper = SecretToBackupEntryMapper(predicates)
+
+			Expect(fakeClient.Create(ctx, backupEntry)).To(Succeed())
+
+			Expect(mapper.Map(ctx, logr.Discard(), fakeClient, secret)).To(BeEmpty())
+		})
+
+		It("should find no objects because list is empty", func() {
+			Expect(mapper.Map(ctx, logr.Discard(), fakeClient, secret)).To(BeEmpty())
+		})
+
+		It("should find no objects because the passed object is not secret", func() {
+			Expect(mapper.Map(ctx, logr.Discard(), fakeClient, configMap)).To(BeEmpty())
+		})
+	})
+
+	Describe("#NamespaceToBackupEntryMapper", func() {
+		var mapper mapper.Mapper
+
+		BeforeEach(func() {
+			mapper = NamespaceToBackupEntryMapper(nil)
+		})
+
+		It("should find all objects for the passed cluster", func() {
```

Comment:
```suggestion
		It("should find all objects for the passed namespace", func() {
```
Adapt similar occurrences as well .